src/topology/is_locally_homeomorph.lean

```diff
@@ -0,0 +1,53 @@
+/-
+Copyright (c) 2021 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+import topology.local_homeomorph
+
+/-!
+# Local homeomorphisms
+
+This file defines local homeomorphisms.
+
+## Main definitions
+
+`is_locally_homeomorph`: A function `f : X → Y` satisfies `is_locally_homeomorph` if it is
+locally a homeomorphism.
+-/
+
+variables {X Y Z : Type*} [topological_space X] [topological_space Y] [topological_space Z]
+  (g : Y → Z) (f : X →  Y)
+
+/-- A function `f : X → Y` satisfies `is_locally_homeomorph` if it is locally a homeomorphism. -/
+def is_locally_homeomorph :=
+∀ x : X, ∃ e : local_homeomorph X Y, f = e ∧ x ∈ e.source
```

Comment:
I think it's worth spelling out in the docstring what this means. For people who don't know what "locally a homeomorphism" means, this sentence is a tautology.

Also, I think it might be useful to have a convenience constructor that only requires `f` and `e` to be equal on some neighbourhood of `x`.